### PR TITLE
Create a database based on a diagram

### PR DESCRIPTION
**Pull request by ogagaoghene**
# Create a database based on a diagram

We created a file named schema_based_on_diagram.sql where we implemented the database from the diagram.
We Joined tables from many-to-many relationships.
added the FK indexes.

clinic_diagram

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+CREATE TABLE invoice_items (
+id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+unit_price DECIMAL,
+quantity INT,
+total_price DECIMAL,
+invoice_id INT,
+treatment_id INT,
+FOREIGN KEY (invoice_id) REFERENCES invoices(id),
+FOREIGN KEY (treatment_id) REFERENCES treatment(id)
+);
```
Comment (gemmen29) [inline schema_based_on_diagram.sql:35]: - [x] Kindly add the table representing the many-to-many relationship between `medical_histories` and `treatment` as per the schema diagram.

Comment (gemmen29) [inline schema_based_on_diagram.sql:35]: - [x] Kindly add indexes for all foreign keys as per this [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/database_from_diagram.md#:~:text=Remember%20to%20add%20the%20FK%20indexes.).

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (Said-laasri) [reply schema_based_on_diagram.sql:35]: @gemmen29  thanks 
fixed.

Comment (Said-laasri) [reply schema_based_on_diagram.sql:35]: @gemmen29  thnaks

fixed .

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+type VARCHAR(50),
+name VARCHAR(50)
+);
+
+CREATE TABLE medical_histories (
+id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+admitted_at TIMESTAMP,
+patient_id INT REFERENCES patients(id),
+status VARCHAR(50)
+);
```
Comment (Omar-Muhamad) [inline schema_based_on_diagram.sql:18]: - [ ] As the previous reviewer mentioned, Please add a table that represents the many-to-many relationship between `medical_histories` and `treatment` and you can call it something like `treatments_histories`. See this [link ](https://hasura.io/learn/database/postgresql/core-concepts/6-postgresql-relationships/#:~:text=have%20multiple%20articles.-,Many%2Dto%2DMany,-A%20many%20to)to understand more about how to address many-to-many relationships.

Comment (omar-muhamad) [inline schema_based_on_diagram.sql:18]: - [x] As the previous reviewer mentioned, Please add a table that represents the many-to-many relationship between `medical_histories` and `treatment` tables and you can call it `treatments-histories`. See this [link](https://hasura.io/learn/database/postgresql/core-concepts/6-postgresql-relationships/#:~:text=have%20multiple%20articles.-,Many%2Dto%2DMany,-A%20many%20to) to understand more about how to address many-to-many relationship.

File: queries.sql [hunk 3]
```diff
@@ -73,4 +73,12 @@ GROUP BY s.name;
...
 EXPLAIN ANALYZE SELECT COUNT(*) FROM visits where animals_id = 4;
 EXPLAIN ANALYZE SELECT * FROM visits where vets_id = 2;
-EXPLAIN ANALYSE SELECT * FROM owners where email = 'owner_18327@mail.com';
\ No newline at end of file
+EXPLAIN ANALYSE SELECT * FROM owners where email = 'owner_18327@mail.com';
+
+--  CREATE INDEXES 
+ CREATE INDEX ON visits (animals_id); 
+ CREATE INDEX ON visits (vet_id); 
+ CREATE INDEX ON owners (email);
```
Comment (omar-muhamad) [inline queries.sql:81]: - [x] As the previous reviewer mentioned, Please add indexes for all foreign keys as per this [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/database_from_diagram.md#:~:text=Remember%20to%20add%20the%20FK%20indexes.) and kindly remove the indexes from here to the `schema` file just below the tables you created.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @ogagaoghene and @Said-laasri,

Nice job making the required changes. Unfortunately, you missed some of them but you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Github flow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [x] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: schema_based_on_diagram.sql (hunk 2 above)
Comment (Said-laasri) [reply schema_based_on_diagram.sql:18]: @Omar-Muhamad it done
sorry didn't understand at first
rela

File: queries.sql (hunk 3 above)
Comment (Said-laasri) [reply queries.sql:81]: @Omar-Muhamad  thanks 
i fixed it .
indx

**Comment by omar-muhamad** [comment]
### Tip for the next reviewer:
The students pushed the changes after they ask for the second review and I submitted my review without seeing these changes.
I think everything is fixed now.

**Review by KDlamini — APPROVED** [review]
### Hi @ogagaoghene, @Said-laasri 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️ All required SQL well implemented

# Status: Approved ✔️ 

oscars-standing-ovation
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**